### DISCUSSION TITLE
Branch automerge with required review

### DISCUSSION THREAD

**Discussion by B4nan** [renovatebot/renovate#10901] [Request Help]
Hi there, trying to make renovate automerging work with branches instead of PRs. As far as I understood, it won't work with required PR reviews, it even creates the PR and says it was not able to automerge (https://github.com/B4nan/lerna-test-v1/pull/13). 

Do I understand correctly that this can't be gotten around with the https://github.com/apps/renovate-approve bot, as that works only for real PRs and not branches?

I thought there could be easy workaround to this by disabling required reviews and handling that via another GH action/check, which would be green only if there is at least one review done - and such check would be present only for real PRs and not branches, therefore it would not be present in what renovate sees, only for real PRs made by other developers. Is this a valid idea? Does such a GH action exist already? Tried to find something without any luck so far, but it feels easy enough to write a custom one (although I have close to zero experience with GH api, so would be great to find something similar for reference). 

The repo https://github.com/B4nan/lerna-test-v1 is a testing one, where the branch automerge worked before I enabled required reviews. The real repo where I am trying to make this work is https://github.com/apify/apify-shared-js, and in that one it won't even create the `all-non-major` branch, which is another weird issue I am facing right now. 

latest logs:
https://app.renovatebot.com/dashboard#github/B4nan/lerna-test-v1/406254287
https://app.renovatebot.com/dashboard#github/apify/apify-shared-js/406254327

**Comment by rarkins** [comment]
> Do I understand correctly that this can't be gotten around with the [github.com/apps/renovate-approve]
(https://github.com/apps/renovate-approve) bot, as that works only for real PRs and not branches?

You understand correctly. Required PR reviews in GitHub implicitly means "PRs are required".
 
> I thought there could be easy workaround to this by disabling required reviews and handling that via another GH action/check, which would be green only if there is at least one review done - and such check would be present only for real PRs and not branches, therefore it would not be present in what renovate sees, only for real PRs made by other developers. Is this a valid idea? 

Semi-valid. Remember that GitHub enforces that the required reviewer has appropriate permissions to commit, e.g. that it's not some internet rando or intern about to ruin your day by approving something that's not meant to be approved. If you work in a controlled environment where *anyone* who might be approving is authorized to then you don't need to worry, but otherwise you'd need more advanced logic to detect *who* is allowed to review.

> Does such a GH action exist already? Tried to find something without any luck so far, but it feels easy enough to write a custom one (although I have close to zero experience with GH api, so would be great to find something similar for reference).

I have no idea but I think that anything which duplicates what GitHub already does is less likely to be implemented already
